core/orchestrator.go

```diff
@@ -515,19 +509,18 @@ type RemoteTranscoderManager struct {
 	RTmutex           *sync.Mutex
 }
 
-func (rtm *RemoteTranscoderManager) Register(transcoder *RemoteTranscoder) {
+func (rtm *RemoteTranscoderManager) Manage(transcoder *RemoteTranscoder) {
 	rtm.RTmutex.Lock()
-	defer rtm.RTmutex.Unlock()
 	rtm.liveTranscoders[transcoder.stream] = transcoder
-
 	for i := 0; i < transcoder.Capacity(); i++ {
 		rtm.remoteTranscoders = append(rtm.remoteTranscoders, transcoder)
 	}
-}
+	rtm.RTmutex.Unlock()
+
+	<-transcoder.eof
```

Comment:
Much more succinct way of building the `Register` and `Unregister` functions. I didn't think about using `<-transcoder.eof` as a blocker during this function (and have the rest of the function not execute), but I see how that works now.

Reply:
I didn't think of it either until noticed that the API felt clunky while writing the transcoding tests for the manager. That prompted a closer review, and here we are.